### PR TITLE
Add argument for validation sample count & update source to validation set

### PR DESCRIPTION
**Pull request by akrentz6**
This commit adds a `--val_split` argument (default = 50) to `arn_generic.py` to allow users to explicitly set the number of validation samples used during training. 

Previously, the code automatically used the same number of samples for both training and validation by duplicating the sample count and took all of these samples from the training set, leaving the validation set from `mnist.npz` unused.

Now, users can choose to train on the entire 60,000 image dataset (with 10,000 images for validation) whilst the default still remains at 50 images for training and validation each (now taken from the correct dataset).

### CODE REVIEW

**Comment by markgarnold** [comment]
How much have you tested this?

I also have a concern about the last PR from you about argmax: GitHub attributed this to Nicolas Brunie (an established PhD researcher in computer arithmetic--I don't know him personally, but he is coauthor with French researchers that I do know).  Do you know Nicolas, or share some other repos with him or what?   

I am guessing this has something to do with the request from @pradeeban not to incorporate thousands of lines of indentation changes together with substantive changes.

**Comment by akrentz6** [comment]
I've tested the new argument with a variety of different combinations of arguments and they all work as expected - the change is relatively simple, mostly just changing the declarations of `x_val` and `y_val` to be from `x_test` and `y_test` rather than the training data - both `x_test` and `y_test` were already declared but weren't being used.

> I also have a concern about the last PR from you about argmax: GitHub attributed this to Nicolas Brunie (an established PhD researcher in computer arithmetic--I don't know him personally, but he is coauthor with French researchers that I do know). Do you know Nicolas, or share some other repos with him or what?

In terms of this I'm quite confused. My previous PR changed very few lines and only involves a very common use of `**kwargs`. I'm not familiar with Nicolas Brunie and don't know him.

> I am guessing this has something to do with the request from @pradeeban not to incorporate thousands of lines of indentation changes together with substantive changes.

I do apologise for this, there's an inconsistent use of tabs and spaces in `arn_generic.py` (most of the code is tab indented whilst the if statements for different xlns objects are spaced) and when I cloned the repository and ran `arn_generic.py`, my IDE changed it all to spaces as python throws an error when there's both.

When I submitted the pull request I thought it would be fine to leave this fix in as I haven't had much experience contributing to other projects and wasn't really familiar with pull request standards. It won't happen again, my mistake.

**Comment by markgarnold** [comment]
Sorry for my confusion, @akrentz6 .  I don't know why you are not mentioned as a contributor on the main page, but it is clear that your contributions are yours.  Nicolas contributed a typo correction several weeks ago, which has nothing to do with what you contributed yesterday.  I find that GitHub is a bit opaque at times...

**Comment by akrentz6** [comment]
@markgarnold No worries at all, thanks for reviewing these PRs.

**Comment by pradeeban** [comment]
Thanks, @akrentz6!

Yes, "GitHub AI" can be strange at times. That is why I recommend that the commits be minimal and not include irrelevant changes, such as those style fixes in the first (closed without merging) pull request from @akrentz6. However, the subsequent pull requests from @akrentz6 are of high quality, and I have no complaints.

Good work.
